### ISSUE TITLE
withOSDiskStorageAccountType does not work in VirtualMachineScaleSet

### ISSUE THREAD

**Issue by tgc123** [Azure/azure-sdk-for-java#1921]
Setting OS disk type to PREMIUM_LRS does not affect OS disk type. 

Having the following call in the code causes the request below:

Call:
`.withOSDiskStorageAccountType(StorageAccountTypes.PREMIUM_LRS)`

Request:
`
       "storageProfile": {
         "osDisk": {
           "createOption": "FromImage",
           "caching": "ReadWrite",
           "managedDisk": {
             "storageAccountType": "Standard_LRS"
           }
         },`

In the code level, it seems that .withOSDiskStorageAccountType() does not change the OSDisk type, but instead it changes the default storage account type for datadisks. The code is the same as for .withDataDiskDefaultStorageAccountType():

```
   public VirtualMachineScaleSetImpl withOSDiskStorageAccountType(StorageAccountTypes accountType) {
        this.managedDataDisks.setDefaultStorageAccountType(accountType);
        return this;
    }
```

```
    public VirtualMachineScaleSetImpl withDataDiskDefaultStorageAccountType(StorageAccountTypes storageAccountType) {
        this.managedDataDisks.setDefaultStorageAccountType(storageAccountType);
        return this;
    }
```

**Comment by tgc123** [comment]
As a workaround, it's possible to call this after VirtualMachineScaleSets().define(), "vmss" is the output from define():

`((VirtualMachineScaleSetImpl)vmss).inner().virtualMachineProfile().storageProfile().osDisk().withManagedDisk(new VirtualMachineScaleSetManagedDiskParameters().withStorageAccountType(StorageAccountTypes.PREMIUM_LRS));`

**Comment by anuchandy** [comment]
Thanks @tgc123, it's a bug, assigning to me.

**Comment by martinsawicki** [comment]
fixed in v1.4
